src/libraries/System.Private.CoreLib/src/System/BitConverter.cs

```diff
@@ -159,6 +159,33 @@ public static bool TryWriteBytes(Span<byte> destination, long value)
             return true;
         }
 
+        /// <summary>
+        /// Returns the specified 128-bit signed integer value as an array of bytes.
+        /// </summary>
+        /// <param name="value">The number to convert.</param>
+        /// <returns>An array of bytes with length 16.</returns>
+        public static byte[] GetBytes(Int128 value)
+        {
+            byte[] bytes = new byte[Int128.Size];
+            Unsafe.As<byte, Int128>(ref bytes[0]) = value;
+            return bytes;
+        }
+
+        /// <summary>
+        /// Converts a 128-bit signed integer into a span of bytes.
+        /// </summary>
+        /// <param name="destination">When this method returns, the bytes representing the converted 128-bit signed integer.</param>
+        /// <param name="value">The 128-bit signed integer to convert.</param>
+        /// <returns><see langword="true"/> if the conversion was successful; <see langword="false"/> otherwise.</returns>
+        public static bool TryWriteBytes(Span<byte> destination, Int128 value)
+        {
+            if (destination.Length < Int128.Size)
+                return false;
+
+            Unsafe.WriteUnaligned(ref MemoryMarshal.GetReference(destination), value);
```

Comment:
Why use WriteUnaligned here and Unsafe.As in the array-based overload? Is the array-based one depending on the runtime guaranteeing alignment of arrays?

Reply:
Just copy/pasting the existing pattern used by all the other APIs.

It could be that it's relying on runtime alignment guarantees and it could also be that someone didn't consider alignment for arrays. We aren't super consistent about either throughout the BCL.